libretroshare/src/rsserver/p3peers.cc

```diff
@@ -1595,13 +1612,24 @@ std::string p3Peers::GetRetroshareInvite( const RsPeerId& sslId, RetroshareInvit
 
 	if (getPeerDetails(ssl_id, detail))
 	{
-        if(!(invite_flags & RetroshareInviteFlags::FULL_IP_HISTORY) || detail.isHiddenNode)
-            detail.ipAddressList.clear();
+		if(!(invite_flags & RetroshareInviteFlags::FULL_IP_HISTORY) || detail.isHiddenNode)
+			detail.ipAddressList.clear();
+
+		//Check if external address is IPv6, then move it to ipAddressList as RsCertificate only allow 4 numbers.
+		struct in6_addr sin6_addr;
+		if( inet_pton(AF_INET6, detail.extAddr.c_str(), &(sin6_addr))
```

Comment:
Same as before use our wrappers instead of calling system functions directly

Reply:
Done